include/core.hpp

```diff
@@ -33,7 +33,10 @@ namespace color_coded
 
     inline std::string temp_dir()
     {
-      static auto const dir(fs::temp_directory_path() / "color_coded/");
+      static auto const temp_dir(fs::temp_directory_path());
+      static auto const user_dir(std::string("color_coded-") +
+                                 std::to_string(geteuid()));
```

Comment:
Nice, we've avoided the nullptr issue. We still have the change in behavior though, since the old path used to end in a slash and the new path does not. Will you add on the trailing slash, to keep things consistent? Otherwise, the code which uses `temp_dir` will not actually go into the `color_coded` directory.